documentation/src/main/asciidoc/topics/con_jvm_heap_memory_allocation.adoc

```diff
@@ -1,26 +1,100 @@
 [id='jvm-heap-memory-allocation_{context}']
 = JVM heap space allocation
 
-When allocating JVM heap space for {brandname} deployments, you can use the following guidelines:
+Determine the amount of JVM memory that is required for your {brandname} deployment, so that you have sufficient data storage capacity to meet your needs.
+
+[IMPORTANT]
+====
+Allocating too much of JVM heap space for data storage can lead to more frequent garbage collection (GC) and GC pauses.
```

Comment:
I would change to somethink like:
Improper combination of JavaHeap size and GC settingcan lead to more frequent GC or less GC runs with a very long full stop pause.

Reply:
Updated to:

> Allocating a large amount of memory heap size in combination with setting garbage collection (GC) times can cause either of the following behavior changes for your application:
> 
> * More GC runs that reduces the application's performance.
> * Less GC runs that lead to longer GC pauses.

I assume the application's performance is affected by this combination? Or would it be the JVM? What about off-heap memory?

Reply:
If the GC is running in parallel to mark dead objects it is 'only' a matter of CPU power and the application might be not or low affected.
If there is no free or fragmented memory if the JVM try to allocate it will run the GC in a 'stop the world' manner which mean the application will not respond for that time. Such pause can be up to minutes (the larger the memory), this means for JGroups cluster with default after 60sec the node is kicked out of the cluster with all drawbacks of rebalancing etc...

OffHeap is not working like this, but there is no limit for allocation (if there is no eviction configured) it will re-use the freed memory but allocate new from the system if necessary. This will end in a system failure (or pod restart)

Reply:
Thanks, @wfink . I updated this section. Let me know what you think.

Reply:
I don't see what has changed

Reply:
My apologies. The changes did not get pushed. Here is my proposed update:

> Allocating a large memory heap size in combination with setting garbage collection (GC) times might impact the performance of your {brandname} deployment in the following ways:
> * If a JVM can only allocate fragmented memory or low volumes of memory from storage, the GC might operate ahead of the deployment. This aynchronous behavior might cause large GC pauses.
> * If CPU resources are low and GC operates syncrhonously with the deployment, GC might require more frequent runs that can degrade your deployment's performance.

Reply:
JVM does not allocate (fragmented) memory - only allocate memory if -Xms != -Xmx (recommendation is -Xms==-Xmx) and it still not reach -Xmx, it will use the free slots inside of the heap if possible.
But if the heap is fragmented and there is no space for the object a FullGC will run and stop the JVM processing.

Also a typo syncrhonously

Reply:
Updated.